runtime/elem/builtins/Core.h

```diff
@@ -19,28 +20,20 @@ namespace elem
             return channelIndex.load();
         }
 
-        FloatType getTargetGain()
-        {
-            return targetGain.load();
-        }
-
         bool active()
         {
-            return targetGain.load() > 0.0f;
+            return fadeOut.targetGain > 0.5;
         }
 
         bool stillRunning()
         {
-            auto const t = targetGain.load();
-            auto const c = currentGain.load();
-
-            return (t >= 0.5 || (std::abs(c - t) >= 1e-6));
+            return active() || !fadeOut.settled();
         }
 
-        void activate(FloatType initialGain = FloatType(0)) 
+        void setFades(FloatType fadeInMs, FloatType fadeOutMs) 
         {
-            setProperty("active", true);
-            currentGain.store(initialGain);
+            fadeIn.setFadeTimeMs(GraphNode<FloatType>::getSampleRate(), fadeInMs);
```

Comment:
I think it'd be cleaner to still have 1 fade but parameterize the GainFade class to accept different values for up/down steps

Reply:
I had that thought as well, but thought it might get weird if you activate and deactivate quickly. i'll think through that again

Reply:
done!